x-pack/plugins/ingest_manager/server/services/datasource.ts

```diff
@@ -104,8 +104,28 @@ class DatasourceService {
     return (await this.get(soClient, id)) as Datasource;
   }
 
-  public async delete(soClient: SavedObjectsClientContract, id: string): Promise<void> {
-    await soClient.delete(SAVED_OBJECT_TYPE, id);
+  public async delete(
+    soClient: SavedObjectsClientContract,
+    ids: string[]
+  ): Promise<DeleteDatasourcesResponse> {
+    const result: DeleteDatasourcesResponse = [];
+
+    for (const id of ids) {
+      try {
+        await soClient.delete(SAVED_OBJECT_TYPE, id);
```

Comment:
could you promise.all instead?

Reply:
Probably. I don't believe these _have_ to be done sequentially. I copied the implementation from the AgentConfigService.